src/opm/parser/eclipse/EclipseState/Schedule/Action/ActionResult.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+  Copyright 2019 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <vector>
+
+#include <opm/parser/eclipse/EclipseState/Schedule/Action/ActionResult.hpp>
+
+namespace Opm {
+namespace Action {
+
+Result::Result(bool result_arg) :
+    m_result(result_arg)
+{}
+
+Result::Result(bool result_arg, const std::vector<std::string>& wells) :
+    m_result(result_arg),
+    matching_wells(wells)
+{}
+
+Result::Result(bool result_arg, const WellSet& wells) :
+    m_result(result_arg),
+    matching_wells(wells)
+{}
+
+bool Result::result() const {
+    return this->m_result;
+}
+
+std::vector<std::string> Result::wells() const {
+    return this->matching_wells.wells();
+}
+
+bool Result::scalar() const {
+    return matching_wells.scalar();
+}
+
+
+Result& Result::operator|=(const Result& other) {
+    this->m_result |= other.m_result;
```

Comment:
Bitwise 'or' is not defined for `bool`.  You need to say
```C++
this->m_result = this->m_result || other.m_result;
```

Reply:
I believe you - of course - and will update, but the current code compiles without warning and works as intended, maybe that is just an undefined fallback to the whatever underlying integral type?